game/src/main/scala/io/aigar/game/Player.scala

```diff
@@ -73,6 +74,23 @@ class Player(val id: Int, startPosition: Vector2) extends EntityContainer
     grid.randomPosition
   }
 
+  def merge(cells: List[Cell], player: Player): List[Cell] ={
+    var cellsReturn = List[Cell]()
+
+    for (cell <- cells) {
+      for (secondCell <- cells.filterNot(_.id == cell.id)) {
+        if (cell.id != secondCell.id
+          && cell.overlaps(secondCell)
+          && !cellsReturn.contains(cell)) {
+          logger.info(s"Player ${player.id}'s ${cell.id} (mass ${cell.mass}) merged with ${secondCell.id} (mass ${secondCell.mass})")
+          cell.mass += secondCell.mass
+          cellsReturn ::= secondCell
+        }
+      }
+    }
+    cells diff cellsReturn
```

Comment:
I changed the name to be more self-explanatory... It is to return the original list without the cells that have been eaten.